### PR TITLE
fix(ext/node): Support private EC key signing

### PR DESCRIPTION
**Pull request by littledivy**
Fixes https://github.com/denoland/deno/issues/18972

Support for web-push VAPID keys & jws signing

- Fixes EC keygen to return raw private key and uncompressed public key point.
- Support for `EC PRIVATE KEY`

### CODE REVIEW

File: ext/node/ops/crypto/mod.rs [hunk 1]
```diff
@@ -419,6 +435,25 @@ pub fn op_node_sign(
         .into(),
       )
     }
+    // signature structure encoding is DER by default for DSA and ECDSA.
+    //
+    // TODO(@littledivy): Validate public_key if present
```
Comment (bartlomieju) [inline ext/node/ops/crypto/mod.rs:440]: Can you open an issue about it?

File: ext/node/ops/crypto/mod.rs [hunk 2]
```diff
@@ -704,30 +739,32 @@ pub async fn op_node_dsa_generate_async(
 fn ec_generate(
   named_curve: &str,
 ) -> Result<(ToJsBuffer, ToJsBuffer), AnyError> {
-  use ring::signature::EcdsaKeyPair;
-  use ring::signature::KeyPair;
+  use elliptic_curve::sec1::ToEncodedPoint;
 
-  let curve = match named_curve {
+  let mut rng = rand::thread_rng();
+  // TODO(@littledivy): Support public key point encoding.
```
Comment (bartlomieju) [inline ext/node/ops/crypto/mod.rs:745]: Ditto

File: ext/node/ops/crypto/mod.rs (hunk 1 above)
Comment (littledivy) [reply ext/node/ops/crypto/mod.rs:440]: https://github.com/denoland/deno/issues/22919

File: ext/node/ops/crypto/mod.rs (hunk 2 above)
Comment (littledivy) [reply ext/node/ops/crypto/mod.rs:745]: https://github.com/denoland/deno/issues/22919

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**